hyper-lib/src/node.rs

```diff
@@ -480,46 +479,57 @@ impl Node {
     /// Reconciliation requests, as opposed to INVs, are sent on a schedule, no matter if we have transactions to reconcile or not
     /// (since our peer may have and we are unaware of this).
     /// Schedule the next reconciliation request.
-    pub fn schedule_set_reconciliation(&mut self, request_time: u64) -> ScheduledEvent {
-        // Make it so we reconcile with all peers every RECON_REQUEST_INTERVAL
-        let delta = ((RECON_REQUEST_INTERVAL as f64 / self.out_peers.len() as f64)
-            * SECS_TO_NANOS as f64)
-            .round() as u64;
+    pub fn schedule_set_reconciliation(
+        &self,
+        peer_id: &NodeId,
+        request_time: u64,
+    ) -> ScheduledEvent {
         ScheduledEvent::new(
-            Event::process_scheduled_reconciliation(self.node_id),
-            request_time + delta,
+            Event::process_scheduled_reconciliation(self.node_id, *peer_id),
+            request_time,
         )
     }
 
-    /// Processes a previously scheduled reconciliation request. This starts the set reconciliation message exchange
+    /// Processes a previously scheduled reconciliation request. This starts the set reconciliation message exchange.
+    /// This creates a recurring reconciliation schedule for a given peer as long as the transaction has not been announced
+    /// over that link. Afterwards the scheduling is dropped.
     pub fn process_scheduled_reconciliation(
         &mut self,
+        peer_id: &NodeId,
         request_time: u64,
-    ) -> (ScheduledEvent, ScheduledEvent) {
+    ) -> Vec<ScheduledEvent> {
         // Peers are selected for set reconciliation in a round robin manner. This is managed internally by our IndexedMap
-        let (next_peer_id, next_peer) = self.out_peers.get_next();
         debug_log!(
             request_time,
             self.node_id,
-            "Requesting transaction reconciliation to peer_id: {next_peer_id}",
+            "Requesting transaction reconciliation to peer_id: {peer_id}",
         );
 
-        let to_be_reconciled = next_peer
-            .get_tx_reconciliation_state()
-            .unwrap()
-            .get_recon_set();
-
-        // Return two events, one starting the reconciliation flow for the given peer, and another one
-        // scheduling the next reconciliation request (for the next peer in line)
-        (
-            self.send_message_to(
-                NetworkMessage::REQRECON(to_be_reconciled),
-                next_peer_id,
-                request_time,
-            )
-            .unwrap(),
-            self.schedule_set_reconciliation(request_time),
-        )
+        let mut events = Vec::new();
+        let peer = self.get_peer(peer_id).unwrap();
+        let recon_state = peer.get_tx_reconciliation_state().unwrap();
+
+        // Drop the periodic reconciliation if the transaction has already been announced though this link
+        if !(self.knows_transaction() && peer.already_announced()) {
```

Comment:
is it possible to have peer.already_announced but !self.knows_transaction?

If not, just assume the latter instead of conditioning.

If yes... what we do then?

Reply:
It is possible. It could be the case that we received the announcement but not the transaction yet, so `peer.they_announced_tx()` is true.

In this case we do not need to go through reconciliation, since the INV has already been received, but the GETDATA may still be delayed